common/achievement/index.ts

```diff
@@ -171,8 +171,21 @@ async function rewardUser<ID extends ActionID>(evaluationResult: number, userAch
         select: { id: true, userId: true, achievedAt: true, template: true },
     });
 
-    await actionTakenAt(new Date(event.at), event.user, 'user_achievement_reward_issued', {
-        achievement: { name: updatedAchievement.template.name, id: updatedAchievement.id.toString() },
-    });
+    const { type, group, groupOrder } = updatedAchievement.template;
+
+    if (type === achievement_type_enum.SEQUENTIAL) {
+        const templatesByGroup = await getAchievementTemplates(TemplateSelectEnum.BY_GROUP);
+        const groupTemplates = templatesByGroup.get(group);
+        const lastTemplate = groupTemplates[groupTemplates.length - 2];
```

Comment:
So these are sequential achievements. The last template that actually needs evaluation is the second last in the group. The real last template needs no evaluation, since it is just there to display the information of a sequential achievement being rewarded. So we check, whether the last achievement to be evaluated is achieved, and if so, the next (reward achievement) is also generated for this sequential achievement.

I highlighted this inside the code now by mentioning it in a comment.

Reply:
🤔 So it's a hack to show a last state?

Reply:
https://github.com/corona-school/project-user/issues/1109